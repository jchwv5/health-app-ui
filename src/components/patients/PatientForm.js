/* eslint-disable no-unused-vars */
/* eslint-disable jsx-a11y/label-has-associated-control */
import React, { useState } from 'react';
import './PatientForm.css';
import FormItem from '../../utils/FormItem';
import handlePatientApiCall from './HandlePatientApiCall';

/**
 * Modular form to be used for adding new patients and updating existing patients
 * @param {*} props - takes in parameters from onClick that calls the form to determine whether
 * form will be used for adding a new patient or updating an existing one
 * @returns - form component
 */
const PatientForm = (props) => {
  const { onClose } = props;
  const { patientApiCall } = props;
  const { patientId } = props;
  const { formHeader } = props;
  const { buttonLabel } = props;
  const { successMsg } = props;
  const [patientData, setPatientData] = useState({});
  const [apiError, setApiError] = useState(false);
  const [patientSuccess, setPatientSuccess] = useState(false);
  const [patientFailure, setPatientFailure] = useState(false);
  const [patientErrors, setPatientErrors] = useState({
    firstName: { dataIsValid: false, errorMessage: '' },
    lastName: { dataIsValid: false, errorMessage: '' },
    ssn: { dataIsValid: false, errorMessage: '' },
    email: { dataIsValid: false, errorMessage: '' },
    street: { dataIsValid: false, errorMessage: '' },
    city: { dataIsValid: false, errorMessage: '' },
    state: { dataIsValid: false, errorMessage: '' },
    postal: { dataIsValid: false, errorMessage: '' },
    age: { dataIsValid: false, errorMessage: '' },
    height: { dataIsValid: false, errorMessage: '' },
    weight: { dataIsValid: false, errorMessage: '' },
    insurance: { dataIsValid: false, errorMessage: '' },
    gender: { dataIsValid: false, errorMessage: '' }
  });

  const handleChange = (e) => {
    setPatientData((prevValue) => ({ ...prevValue, [e.target.id]: e.target.value }));
  };

  return (
    <div className="formPopup" id="myForm">
      <form action="/action_page.php" className="formContainer">
        <h1 className="formHeader">{formHeader}</h1>
        <div className="form">
          <div className="column">
            <FormItem
              className="input"
              type="text"
              id="firstName"
              label="First Name"
              placeholder="ex. John"
              onChange={handleChange}
              value={patientData.firstName}
            />
            <div className="errorDiv">
              {patientErrors.firstName.dataIsValid === false && (
              <p className="errorMsg">{patientErrors.firstName.errorMessage}</p>
              )}
            </div>
            <FormItem
              className="input"
              type="text"
              id="lastName"
              label="Last Name"
              placeholder="ex. Smith"
              onChange={handleChange}
              value={patientData.lastName}
            />
            <div className="errorDiv">
              {patientErrors.lastName.dataIsValid === false && (
              <p className="errorMsg">{patientErrors.lastName.errorMessage}</p>
              )}
            </div>
            <FormItem
              className="input"
              type="text"
              id="ssn"
              label="SSN"
              placeholder="ex. 555-55-5555"
              onChange={handleChange}
              value={patientData.ssn}
            />
            <div className="errorDiv">
              {patientErrors.ssn.dataIsValid === false && (
              <p className="errorMsg">{patientErrors.ssn.errorMessage}</p>
              )}
            </div>
            <FormItem
              className="input"
              type="text"
              id="email"
              label="Email"
              placeholder="ex. john.smith@gmail.com"
              onChange={handleChange}
              value={patientData.email}
            />
            <div className="errorDiv">
              {patientErrors.email.dataIsValid === false && (
              <p className="errorMsg">{patientErrors.email.errorMessage}</p>
              )}
            </div>
          </div>
          <div className="column">
            <FormItem
              className="input"
              type="text"
              id="street"
              label="Street"
              placeholder="123 Test Street"
              onChange={handleChange}
              value={patientData.street}
            />
            <div className="errorDiv">
              {patientErrors.street.dataIsValid === false && (
              <p className="errorMsg">{patientErrors.street.errorMessage}</p>
              )}
            </div>
            <FormItem
              className="input"
              type="text"
              id="city"
              label="City"
              placeholder="ex. Denver"
              onChange={handleChange}
              value={patientData.city}
            />
            <div className="errorDiv">
              {patientErrors.city.dataIsValid === false && (
              <p className="errorMsg">{patientErrors.city.errorMessage}</p>
              )}
            </div>
            <FormItem
              className="input"
              type="text"
              id="state"
              label="State"
              placeholder="ex. CO"
              onChange={handleChange}
              value={patientData.state}
            />
            <div className="errorDiv">
              {patientErrors.state.dataIsValid === false && (
              <p className="errorMsg">{patientErrors.state.errorMessage}</p>
              )}
            </div>
            <FormItem
              className="input"
              type="text"
              id="postal"
              label="Zip Code"
              placeholder="ex. 55555-5555"
              onChange={handleChange}
              value={patientData.postal}
            />
            <div className="errorDiv">
              {patientErrors.postal.dataIsValid === false && (
              <p className="errorMsg">{patientErrors.postal.errorMessage}</p>
              )}
            </div>
          </div>
          <div className="column">
            <FormItem
              className="input"
              type="text"
              id="age"
              label="Age"
              placeholder="ex. 54"
              onChange={handleChange}
              value={patientData.age}
            />
            <div className="errorDiv">
              {patientErrors.age.dataIsValid === false && (
              <p className="errorMsg">{patientErrors.age.errorMessage}</p>
              )}
            </div>
            <FormItem
              className="input"
              type="text"
              id="height"
              label="Height (in Inches)"
              placeholder="ex. 72"
              onChange={handleChange}
              value={patientData.height}
            />
            <div className="errorDiv">
              {patientErrors.height.dataIsValid === false && (
              <p className="errorMsg">{patientErrors.height.errorMessage}</p>
              )}
            </div>
            <FormItem
              className="input"
              type="text"
              id="weight"
              label="Weight (in Pounds)"
              placeholder="ex. 195"
              onChange={handleChange}
              value={patientData.weight}
            />
            <div className="errorDiv">
              {patientErrors.weight.dataIsValid === false && (
              <p className="errorMsg">{patientErrors.weight.errorMessage}</p>
              )}
            </div>
            <FormItem
              className="input"
              type="text"
              id="insurance"
              label="Insurance Provider"
              placeholder="ex. Blue Cross"
              onChange={handleChange}
              value={patientData.insurance}
            />
            <div className="errorDiv">
              {patientErrors.insurance.dataIsValid === false && (
              <p className="errorMsg">{patientErrors.insurance.errorMessage}</p>
              )}
            </div>
            <FormItem
              className="input"
              type="text"
              id="gender"
              label="Gender"
              placeholder="Enter Male, Female or Other"
              onChange={handleChange}
              value={patientData.gender}
            />
            <div className="errorDiv">
              {patientErrors.gender.dataIsValid === false && (
              <p className="errorMsg">{patientErrors.gender.errorMessage}</p>
              )}
            </div>
          </div>
        </div>
      </form>
      <div className="buttonWrapper">
        <div className="serverMsgDiv">
          {patientSuccess === true && (
          <p className="successMsg">{successMsg}</p>
          )}
          {patientFailure === true && (
          <p className="failureMsg">Oops! Something went wrong!</p>
          )}
        </div>
        <button
          type="submit"
          className="btn"
          onClick={() => handlePatientApiCall(patientData,
            patientId,
            setPatientSuccess,
            setPatientFailure,
            patientErrors,
            setPatientErrors,
            setApiError,
            patientApiCall)}
        >
          {buttonLabel}
        </button>
        <button type="button" className="btn cancel" onClick={onClose}>Close</button>
      </div>
    </div>

  );
};

export default PatientForm;
