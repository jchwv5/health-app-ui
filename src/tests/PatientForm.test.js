import { fireEvent, render } from '@testing-library/react';
import React from 'react';
import ReactDom from 'react-dom';
import renderer from 'react-test-renderer';
import PatientForm from '../components/patients/PatientForm';

it('renders successfully', () => {
  const div = document.createElement('div');
  ReactDom.render(<PatientForm />, div);
  ReactDom.unmountComponentAtNode(div);
});

it('App component matches snapshot', () => {
  const component = renderer.create(<PatientForm />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it('form validation fails empty form', () => {
  const { getByLabelText, getByText } = render(<PatientForm buttonLabel="Edit Patient" />);
  fireEvent.change(getByLabelText('First Name'), { target: { value: '' } });
  fireEvent.change(getByLabelText('Last Name'), { target: { value: '' } });
  fireEvent.change(getByLabelText('SSN'), { target: { value: '' } });
  fireEvent.change(getByLabelText('Email'), { target: { value: '' } });
  fireEvent.change(getByLabelText('Street'), { target: { value: '' } });
  fireEvent.change(getByLabelText('City'), { target: { value: '' } });
  fireEvent.change(getByLabelText('State'), { target: { value: '' } });
  fireEvent.change(getByLabelText('Zip Code'), { target: { value: '' } });
  fireEvent.change(getByLabelText('Age'), { target: { value: '' } });
  fireEvent.change(getByLabelText('Height (in Inches)'), { target: { value: '' } });
  fireEvent.change(getByLabelText('Weight (in Pounds)'), { target: { value: '' } });
  fireEvent.change(getByLabelText('Insurance Provider'), { target: { value: '' } });
  fireEvent.change(getByLabelText('Gender'), { target: { value: '' } });
  fireEvent.click(getByText('Edit Patient'));

  expect(getByText('Gender field must not be left empty',
    'Insurance provider field must not be left empty',
    'Weight field must not be left empty',
    'Height field must not be left empty',
    'Age field must not be left empty',
    'Zip code field must not be left empty',
    'State field must not be left empty',
    'City field must not be left empty',
    'Street field must not be left empty',
    'Email field must not be left empty',
    'SSN field must not be left empty',
    'Last name field must not be left empty',
    'First name field must not be left empty')).toBeInTheDocument();
});

it('form fails invalid Email Field', () => {
  const { getByLabelText, getByText } = render(<PatientForm buttonLabel="Edit Patient" />);
  fireEvent.change(getByLabelText('First Name'), { target: { value: '55' } });
  fireEvent.change(getByLabelText('Last Name'), { target: { value: 'Smith' } });
  fireEvent.change(getByLabelText('SSN'), { target: { value: '555-55-5555' } });
  fireEvent.change(getByLabelText('Email'), { target: { value: 'john.smith@gmail.com' } });
  fireEvent.change(getByLabelText('Street'), { target: { value: '123 Test Street' } });
  fireEvent.change(getByLabelText('City'), { target: { value: 'Denver' } });
  fireEvent.change(getByLabelText('State'), { target: { value: 'CO' } });
  fireEvent.change(getByLabelText('Zip Code'), { target: { value: 'ksdfjahskdjhfdaksj' } });
  fireEvent.change(getByLabelText('Age'), { target: { value: '55' } });
  fireEvent.change(getByLabelText('Height (in Inches)'), { target: { value: '72' } });
  fireEvent.change(getByLabelText('Weight (in Pounds)'), { target: { value: '195' } });
  fireEvent.change(getByLabelText('Insurance Provider'), { target: { value: 'Blue Cross' } });
  fireEvent.change(getByLabelText('Gender'), { target: { value: 'Male' } });
  fireEvent.click(getByText('Edit Patient'));

  expect(getByText('Zip code must be in format XXXXX or XXXXX-XXXX')).toBeInTheDocument();
});

it('form validation fails invalid Email Field', () => {
  const { getByLabelText, getByText } = render(<PatientForm buttonLabel="Edit Patient" />);
  fireEvent.change(getByLabelText('First Name'), { target: { value: '55' } });
  fireEvent.change(getByLabelText('Last Name'), { target: { value: 'Smith' } });
  fireEvent.change(getByLabelText('SSN'), { target: { value: '555-55-5555' } });
  fireEvent.change(getByLabelText('Email'), { target: { value: 'kjdsafhkfjhfa' } });
  fireEvent.change(getByLabelText('Street'), { target: { value: '123 Test Street' } });
  fireEvent.change(getByLabelText('City'), { target: { value: 'Denver' } });
  fireEvent.change(getByLabelText('State'), { target: { value: 'CO' } });
  fireEvent.change(getByLabelText('Zip Code'), { target: { value: '12345' } });
  fireEvent.change(getByLabelText('Age'), { target: { value: '55' } });
  fireEvent.change(getByLabelText('Height (in Inches)'), { target: { value: '72' } });
  fireEvent.change(getByLabelText('Weight (in Pounds)'), { target: { value: '195' } });
  fireEvent.change(getByLabelText('Insurance Provider'), { target: { value: 'Blue Cross' } });
  fireEvent.change(getByLabelText('Gender'), { target: { value: 'Male' } });
  fireEvent.click(getByText('Edit Patient'));

  expect(getByText('Please enter a valid e-mail address')).toBeInTheDocument();
});

it('form validation fails invalid SSN Field', () => {
  const { getByLabelText, getByText } = render(<PatientForm buttonLabel="Edit Patient" />);
  fireEvent.change(getByLabelText('First Name'), { target: { value: '55' } });
  fireEvent.change(getByLabelText('Last Name'), { target: { value: 'Smith' } });
  fireEvent.change(getByLabelText('SSN'), { target: { value: 'kjfhdkasjf' } });
  fireEvent.change(getByLabelText('Email'), { target: { value: 'john.smith@gmail.com' } });
  fireEvent.change(getByLabelText('Street'), { target: { value: '123 Test Street' } });
  fireEvent.change(getByLabelText('City'), { target: { value: 'Denver' } });
  fireEvent.change(getByLabelText('State'), { target: { value: 'CO' } });
  fireEvent.change(getByLabelText('Zip Code'), { target: { value: '12345' } });
  fireEvent.change(getByLabelText('Age'), { target: { value: '55' } });
  fireEvent.change(getByLabelText('Height (in Inches)'), { target: { value: '72' } });
  fireEvent.change(getByLabelText('Weight (in Pounds)'), { target: { value: '195' } });
  fireEvent.change(getByLabelText('Insurance Provider'), { target: { value: 'Blue Cross' } });
  fireEvent.change(getByLabelText('Gender'), { target: { value: 'Male' } });
  fireEvent.click(getByText('Edit Patient'));

  expect(getByText('SSN must be in format XXX-XX-XXXX')).toBeInTheDocument();
});

it('form validation fails invalid State Field', () => {
  const { getByLabelText, getByText } = render(<PatientForm buttonLabel="Edit Patient" />);
  fireEvent.change(getByLabelText('First Name'), { target: { value: '55' } });
  fireEvent.change(getByLabelText('Last Name'), { target: { value: 'Smith' } });
  fireEvent.change(getByLabelText('SSN'), { target: { value: '555-55-5555' } });
  fireEvent.change(getByLabelText('Email'), { target: { value: 'john.smith@gmail.com' } });
  fireEvent.change(getByLabelText('Street'), { target: { value: '123 Test Street' } });
  fireEvent.change(getByLabelText('City'), { target: { value: 'Denver' } });
  fireEvent.change(getByLabelText('State'), { target: { value: 'kjdshfksjah' } });
  fireEvent.change(getByLabelText('Zip Code'), { target: { value: '12345' } });
  fireEvent.change(getByLabelText('Age'), { target: { value: '55' } });
  fireEvent.change(getByLabelText('Height (in Inches)'), { target: { value: '72' } });
  fireEvent.change(getByLabelText('Weight (in Pounds)'), { target: { value: '195' } });
  fireEvent.change(getByLabelText('Insurance Provider'), { target: { value: 'Blue Cross' } });
  fireEvent.change(getByLabelText('Gender'), { target: { value: 'Male' } });
  fireEvent.click(getByText('Edit Patient'));

  expect(getByText('State is invalid')).toBeInTheDocument();
});

it('form validation fails invalid Numeric Field', () => {
  const { getByLabelText, getByText } = render(<PatientForm buttonLabel="Edit Patient" />);
  fireEvent.change(getByLabelText('First Name'), { target: { value: 'John' } });
  fireEvent.change(getByLabelText('Last Name'), { target: { value: 'Smith' } });
  fireEvent.change(getByLabelText('SSN'), { target: { value: '555-55-5555' } });
  fireEvent.change(getByLabelText('Email'), { target: { value: 'john.smith@gmail.com' } });
  fireEvent.change(getByLabelText('Street'), { target: { value: '123 Test Street' } });
  fireEvent.change(getByLabelText('City'), { target: { value: 'Denver' } });
  fireEvent.change(getByLabelText('State'), { target: { value: 'CO' } });
  fireEvent.change(getByLabelText('Zip Code'), { target: { value: '12345' } });
  fireEvent.change(getByLabelText('Age'), { target: { value: '55' } });
  fireEvent.change(getByLabelText('Height (in Inches)'), { target: { value: '72' } });
  fireEvent.change(getByLabelText('Weight (in Pounds)'), { target: { value: 'kdsafl;fk;l' } });
  fireEvent.change(getByLabelText('Insurance Provider'), { target: { value: 'Blue Cross' } });
  fireEvent.change(getByLabelText('Gender'), { target: { value: 'Male' } });
  fireEvent.click(getByText('Edit Patient'));

  expect(getByText('Weight should be numbers only')).toBeInTheDocument();
});

it('form validation fails invalid Gender Field', () => {
  const { getByLabelText, getByText } = render(<PatientForm buttonLabel="Edit Patient" />);
  fireEvent.change(getByLabelText('First Name'), { target: { value: 'John' } });
  fireEvent.change(getByLabelText('Last Name'), { target: { value: 'Smith' } });
  fireEvent.change(getByLabelText('SSN'), { target: { value: '555-55-5555' } });
  fireEvent.change(getByLabelText('Email'), { target: { value: 'john.smith@gmail.com' } });
  fireEvent.change(getByLabelText('Street'), { target: { value: '123 Test Street' } });
  fireEvent.change(getByLabelText('City'), { target: { value: 'Denver' } });
  fireEvent.change(getByLabelText('State'), { target: { value: 'CO' } });
  fireEvent.change(getByLabelText('Zip Code'), { target: { value: '12345' } });
  fireEvent.change(getByLabelText('Age'), { target: { value: '55' } });
  fireEvent.change(getByLabelText('Height (in Inches)'), { target: { value: '72' } });
  fireEvent.change(getByLabelText('Weight (in Pounds)'), { target: { value: '195' } });
  fireEvent.change(getByLabelText('Insurance Provider'), { target: { value: 'Blue Cross' } });
  fireEvent.change(getByLabelText('Gender'), { target: { value: 'ksdajfhskd' } });
  fireEvent.click(getByText('Edit Patient'));

  expect(getByText('Gender field should be male, female, or other')).toBeInTheDocument();
});
