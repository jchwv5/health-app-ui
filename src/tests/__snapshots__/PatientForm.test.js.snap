// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
<div
  className="formPopup"
  id="myForm"
>
  <form
    action="/action_page.php"
    className="formContainer"
  >
    <h1
      className="formHeader"
    />
    <div
      className="form"
    >
      <div
        className="column"
      >
        <div>
          <label
            className="label"
            htmlFor="firstName"
          >
            First Name
            <div>
              <input
                className="input"
                id="firstName"
                onChange={[Function]}
                placeholder="ex. John"
                type="text"
              />
            </div>
          </label>
        </div>
        <div
          className="errorDiv"
        >
          <p
            className="errorMsg"
          >
            
          </p>
        </div>
        <div>
          <label
            className="label"
            htmlFor="lastName"
          >
            Last Name
            <div>
              <input
                className="input"
                id="lastName"
                onChange={[Function]}
                placeholder="ex. Smith"
                type="text"
              />
            </div>
          </label>
        </div>
        <div
          className="errorDiv"
        >
          <p
            className="errorMsg"
          >
            
          </p>
        </div>
        <div>
          <label
            className="label"
            htmlFor="ssn"
          >
            SSN
            <div>
              <input
                className="input"
                id="ssn"
                onChange={[Function]}
                placeholder="ex. 555-55-5555"
                type="text"
              />
            </div>
          </label>
        </div>
        <div
          className="errorDiv"
        >
          <p
            className="errorMsg"
          >
            
          </p>
        </div>
        <div>
          <label
            className="label"
            htmlFor="email"
          >
            Email
            <div>
              <input
                className="input"
                id="email"
                onChange={[Function]}
                placeholder="ex. john.smith@gmail.com"
                type="text"
              />
            </div>
          </label>
        </div>
        <div
          className="errorDiv"
        >
          <p
            className="errorMsg"
          >
            
          </p>
        </div>
      </div>
      <div
        className="column"
      >
        <div>
          <label
            className="label"
            htmlFor="street"
          >
            Street
            <div>
              <input
                className="input"
                id="street"
                onChange={[Function]}
                placeholder="123 Test Street"
                type="text"
              />
            </div>
          </label>
        </div>
        <div
          className="errorDiv"
        >
          <p
            className="errorMsg"
          >
            
          </p>
        </div>
        <div>
          <label
            className="label"
            htmlFor="city"
          >
            City
            <div>
              <input
                className="input"
                id="city"
                onChange={[Function]}
                placeholder="ex. Denver"
                type="text"
              />
            </div>
          </label>
        </div>
        <div
          className="errorDiv"
        >
          <p
            className="errorMsg"
          >
            
          </p>
        </div>
        <div>
          <label
            className="label"
            htmlFor="state"
          >
            State
            <div>
              <input
                className="input"
                id="state"
                onChange={[Function]}
                placeholder="ex. CO"
                type="text"
              />
            </div>
          </label>
        </div>
        <div
          className="errorDiv"
        >
          <p
            className="errorMsg"
          >
            
          </p>
        </div>
        <div>
          <label
            className="label"
            htmlFor="postal"
          >
            Zip Code
            <div>
              <input
                className="input"
                id="postal"
                onChange={[Function]}
                placeholder="ex. 55555-5555"
                type="text"
              />
            </div>
          </label>
        </div>
        <div
          className="errorDiv"
        >
          <p
            className="errorMsg"
          >
            
          </p>
        </div>
      </div>
      <div
        className="column"
      >
        <div>
          <label
            className="label"
            htmlFor="age"
          >
            Age
            <div>
              <input
                className="input"
                id="age"
                onChange={[Function]}
                placeholder="ex. 54"
                type="text"
              />
            </div>
          </label>
        </div>
        <div
          className="errorDiv"
        >
          <p
            className="errorMsg"
          >
            
          </p>
        </div>
        <div>
          <label
            className="label"
            htmlFor="height"
          >
            Height (in Inches)
            <div>
              <input
                className="input"
                id="height"
                onChange={[Function]}
                placeholder="ex. 72"
                type="text"
              />
            </div>
          </label>
        </div>
        <div
          className="errorDiv"
        >
          <p
            className="errorMsg"
          >
            
          </p>
        </div>
        <div>
          <label
            className="label"
            htmlFor="weight"
          >
            Weight (in Pounds)
            <div>
              <input
                className="input"
                id="weight"
                onChange={[Function]}
                placeholder="ex. 195"
                type="text"
              />
            </div>
          </label>
        </div>
        <div
          className="errorDiv"
        >
          <p
            className="errorMsg"
          >
            
          </p>
        </div>
        <div>
          <label
            className="label"
            htmlFor="insurance"
          >
            Insurance Provider
            <div>
              <input
                className="input"
                id="insurance"
                onChange={[Function]}
                placeholder="ex. Blue Cross"
                type="text"
              />
            </div>
          </label>
        </div>
        <div
          className="errorDiv"
        >
          <p
            className="errorMsg"
          >
            
          </p>
        </div>
        <div>
          <label
            className="label"
            htmlFor="gender"
          >
            Gender
            <div>
              <input
                className="input"
                id="gender"
                onChange={[Function]}
                placeholder="Enter Male, Female or Other"
                type="text"
              />
            </div>
          </label>
        </div>
        <div
          className="errorDiv"
        >
          <p
            className="errorMsg"
          >
            
          </p>
        </div>
      </div>
    </div>
  </form>
  <div
    className="buttonWrapper"
  >
    <div
      className="serverMsgDiv"
    />
    <button
      className="btn"
      onClick={[Function]}
      type="submit"
    />
    <button
      className="btn cancel"
      type="button"
    >
      Close
    </button>
  </div>
</div>
`;
